### ISSUE TITLE
uninferred dimensions are not an exact divisor of input dimensions

### ISSUE THREAD

**Issue by ma-siddiqui** [NVIDIA-AI-IOT/torch2trt#161]
I am going to convert mask rcnn pytorch model into trt. While converting it I am facing errors at certain points. Please let me know what is wrong.

[TensorRT] ERROR: (Unnamed Layer* 1168) [Shuffle]: uninferred dimensions are not an exact divisor of input dimensions, so inferred dimension cannot be calculated
[TensorRT] ERROR: (Unnamed Layer* 1170) [Fully Connected]: kernel weights has count 12845056 but 1024 was expected
[TensorRT] ERROR: (Unnamed Layer* 1174) [Fully Connected]: kernel weights has count 9216 but 9 was expected

Thanks

**Comment by xieydd** [comment]
same error, have you solved it yet; @ma-siddiqui

**Comment by ma-siddiqui** [comment]
Hi,
I am still struggling to solve but unfortunately no luck.

**Comment by ma-siddiqui** [comment]
If anyone knows root cause of this issue please share.

**Comment by ma-siddiqui** [comment]
I have modified the converter for linear layer. Though the errors are removed but not sure this is correct implementation.

@tensorrt_converter('torch.nn.Linear.forward')
def convert_Linear(ctx):
    module = ctx.method_args[0]
    input = ctx.method_args[1]
    input_trt = trt_(ctx.network, input)
    output = ctx.method_return

    weight = trt_(ctx.network, module.weight)
    layer = ctx.network.add_matrix_multiply(input_trt, trt.MatrixOperation.NONE, weight, trt.MatrixOperation.TRANSPOSE)
    output._trt = layer.get_output(0)

**Comment by xieydd** [comment]
@ma-siddiqui I try the solution, good for me, but when train linear layer, can`t use bias=True.

**Comment by ma-siddiqui** [comment]
I am only dealing with inference and also did not verify the converted model. Will share feedback once training is started.

**Comment by xieydd** [comment]
@ma-siddiqui #191 will solve your problem.
